### PR TITLE
Add ledger api test tool cases for expensive daml maps and lists [DPP-955]

### PR DESCRIPTION
**Pull request by mziolekda**

### CODE REVIEW

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_14/LimitsIT.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.daml.ledger.client.binding.Primitive.Party
+import com.daml.ledger.test.semantic.Limits.{WithList, WithMap}
+
+final class LimitsIT extends LedgerTestSuite {
+
+  test(
+    "LLargeMapInContract",
+    "Create a contract with a field containing large map",
+    allocate(SingleParty),
+    8.0,
```
Comment (skisel-da) [inline ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_14/LimitsIT.scala:17]:
```suggestion
    timeoutScale = 8.0,
```

**Review by skisel-da — APPROVED** [review]

File: ledger/test-common/src/main/daml/semantic/Limits.daml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import DA.Map qualified as M
+
+template WithMap
+  with
+    owner : Party
+    items : M.Map Text Party
+  where
+    signatory owner
+    nonconsuming choice WithMap_Noop : ()
```
Comment (tudor-da) [inline ledger/test-common/src/main/daml/semantic/Limits.daml:14]: what's the purpose of this no-op nonconsuming choice?

Comment (tudor-da) [reply ledger/test-common/src/main/daml/semantic/Limits.daml:14]: if it's for forcing deserialization in during interpretation, isn't it enough to just archive it?

**Comment by tudor-da** [comment]
Looks good. Left a question regarding a simplification.

File: ledger/test-common/src/main/daml/semantic/Limits.daml (hunk 2 above)
Comment (mziolekda) [reply ledger/test-common/src/main/daml/semantic/Limits.daml:14]: Well, when it is a noop you can run it a 1000 times in parallel and that's what you can do in dev experiments based on that contarct

File: ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_14/LimitsIT.scala (hunk 1 above)
Comment (mziolekda) [reply ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_14/LimitsIT.scala]: oops, that should actually not be there

**Pull request merged**